swingset/src/main/java/com/nqadmin/swingset/datasources/RowSetOps.java

```diff
@@ -519,36 +547,51 @@ public static void updateColumnText(final RowSet _rowSet, final String _updatedV
 			break;
 			
 		case TIME:
-// TODO: Probably a better way to handle date to timestamp conversion. Formatter? Get rid of getSQLDate() if possible.
-// 2022-05-31_BP: Probably best to cast to a LocalDateTime and return if that fails.
-//	Can convert to SQL Timestamp if successful.			
-			// CONVERT ANY 10 CHARACTER DATE (e.g., yyyy-mm-dd, mm/dd/yyyy to a date/time)			
-			Time timeValue = java.sql.Time.valueOf(_updatedValue);
-			_rowSet.updateTime(_columnName, timeValue);
+			// TODO: there's an ignored IllegalArgumentException possible
+			// Convert a time like "hh:mm:ss"
+// TODO: Better way to handle Time conversion? Formatter? 
+			if (_updatedValue.length() == 8) {
+				try {
+					Time timeValue = java.sql.Time.valueOf(_updatedValue);
+					_rowSet.updateTime(_columnName, timeValue);
+				} catch(IllegalArgumentException ex) {
+					logger.warn("updateColumnText: TIME: " + ex.getMessage());
+					throw ex;
+				}
+			}
 			break;
 			
 		case TIMESTAMP:
-// TODO: Probably a better way to handle date to timestamp conversion. Formatter? Get rid of getSQLDate() if possible.
+			// TODO: there's an ignored IllegalArgumentException possible
+			// Convert something like
+			// "yyyy-mm-dd hh:mm:ss" or "yyyy-mm-dd hh:mm:ss.f[ff...]"
+
+// TODO: Probably a better way to handle date to timestamp conversion. Formatter?
 // 2022-05-31_BP: Probably best to cast to a LocalDateTime and return if that fails.
-//	Can convert to SQL Timestamp if successful.			
-			// CONVERT ANY 10 CHARACTER DATE (e.g., yyyy-mm-dd, mm/dd/yyyy to a date/time)
-			if (_updatedValue.length() == 10) {
-				Timestamp timestampValue = new Timestamp(SSCommon.getSQLDate(_updatedValue).getTime());
-				_rowSet.updateTimestamp(_columnName, timestampValue);
-			} else {
-			// Per ER email 2020-11-25, we weren't even trying to handle a legitimate timestamp
-				Timestamp timeStampValue = java.sql.Timestamp.valueOf(_updatedValue);
-				_rowSet.updateTimestamp(_columnName, timeStampValue);
+			Timestamp timestampValue;
+			try {
+				if (_updatedValue.length() == 19 || _updatedValue.length() > 20) {
+					//System.err.println("TIMESTAMP update: " + _updatedValue);
+					timestampValue = java.sql.Timestamp.valueOf(_updatedValue);
+					_rowSet.updateTimestamp(_columnName, timestampValue);
+				}
+			} catch(IllegalArgumentException ex) {
+				logger.warn("updateColumnText: TIMESTAMP: " + ex.getMessage());
+				throw ex;
```

Comment:
If only date is entered, string length 10, we should probably support that. Based on the above code database will not be updated if only date is entered.

Reply:
> If only date is entered, string length 10, we should probably support that. Based on the above code database will not be updated if only date is entered.

Why is that a problem?

But OK, I'll put it back.

Makes me wonder why a database column would be a TIMESTAMP and not a DATE if all you needed is date. I don't think TIMESTAMP or TIME is ever used; it looks like they would have gotten an exception on almost every keystroke.

Hopefully the mask formatting stuff can be finished and eventually find it's way into the grid parsing. It's weird that only one date format is accepted.

Reply:
DONE

Reply:
Agree that the column type should be Date in those cases but if the user types in just the date it would be good if we can update the database with that date rather than the update failing, especially because it have been working until now.